internal/tests/README.md

```diff
@@ -77,19 +77,29 @@ test cases.
 
 Each test case should pass in:
 ```go
-type testCase struct {
+type TestCase struct {
```

Comment:
Nit: it might be nice to transition this example code to a [testable example](https://go.dev/blog/examples) that has compile time checks.

The descriptions of each field in the struct here also probably make sense to put in godoc so that there's one source of truth to update.